### PR TITLE
attempt to maintain furthest position byte channel has reached

### PR DESCRIPTION
**Pull request by malbx**
@joshbooks Let me know if this is what you had in mind for the TODO addressed by this PR.

### CODE REVIEW

**Comment by joshbooks** [comment]
It is exactly what I had in mind but I was wrong, looks like the contention on the AtomicLong is a bigger deal than I thought.
before: 
It took 51759670 nanoseconds to read 32 objects, each 65536 bytes long
It took 48412782 nanoseconds to read 32 objects, each 65536 bytes long
It took 42621039 nanoseconds to read 32 objects, each 65536 bytes long
It took 52777367 nanoseconds to read 32 objects, each 65536 bytes long
after: 
It took 56631026 nanoseconds to read 32 objects, each 65536 bytes long
It took 54522207 nanoseconds to read 32 objects, each 65536 bytes long
It took 60950088 nanoseconds to read 32 objects, each 65536 bytes long
It took 66747348 nanoseconds to read 32 objects, each 65536 bytes long
Will add a suggestion in a moment.

**Comment by joshbooks** [comment]
Ok, afer appying the change I'm about to suggest we get:
It took 42028089 nanoseconds to read 32 objects, each 65536 bytes long
It took 45712212 nanoseconds to read 32 objects, each 65536 bytes long
It took 46162561 nanoseconds to read 32 objects, each 65536 bytes long
It took 41785378 nanoseconds to read 32 objects, each 65536 bytes long

File: main/src/org/josh/JoshDB/FileTrie/MergeFile.java [hunk 1]
```diff
@@ -96,6 +96,8 @@
   private final AtomicResizingLongArray sequenceArray;
   private ThreadLocal<Long> fd = new ThreadLocal<>();
 
+  private final AtomicLong byteChannelPosition = new AtomicLong(0);
```
Comment (joshbooks) [inline main/src/org/josh/JoshDB/FileTrie/MergeFile.java]:
```suggestion
  private final Counter byteChannelPosition = new Counter();
```

File: main/src/org/josh/JoshDB/FileTrie/MergeFile.java [hunk 2]
```diff
@@ -305,7 +306,7 @@ private boolean computePageArrComplete()
       // increasing and redoing work isn't a huge deal), then we posistion to the minimum
       // of that and the first block we're aware of for this object. That way we don't have
       // to start over at 0 every time
-      channel.position(0);
+      channel.position(byteChannelPosition.get());
```
Comment (joshbooks) [inline main/src/org/josh/JoshDB/FileTrie/MergeFile.java]:
```suggestion
      channel.position(byteChannelPosition.estimate_get());
```

File: main/src/org/josh/JoshDB/FileTrie/MergeFile.java [hunk 3]
```diff
@@ -825,6 +826,8 @@ int readPage(SeekableByteChannel fileStream, byte[] buffer) throws IOException
       // to get better assurances about validity of page info
       addPageInfo(buffer, pageOffset);
```
Comment (joshbooks) [inline main/src/org/josh/JoshDB/FileTrie/MergeFile.java]:
```suggestion
      if(addPageInfo(buffer, pageOffset))
      {
        byteChannelPosition.add(PIPE_BUF);
      }
```

File: main/src/org/josh/JoshDB/FileTrie/MergeFile.java [hunk 4]
```diff
@@ -825,6 +826,8 @@ int readPage(SeekableByteChannel fileStream, byte[] buffer) throws IOException
...
 
+      byteChannelPosition.accumulateAndGet(fileStream.position(), Math::max);
```
Comment (joshbooks) [inline main/src/org/josh/JoshDB/FileTrie/MergeFile.java]:
```suggestion
```

**Review by joshbooks — CHANGES_REQUESTED** [review]

**Comment by malbx** [comment]
I addressed the issues that were pointed out.  Let me know if you notice anything else.

**Review by joshbooks — APPROVED** [review]

**Pull request merged**